File: src/nodes/control/selector.cpp
#include "selector.h"

Selector::Selector() {
    setup();
}

// init the node
void Selector::setup() {
    title = "Selector";
    desc = "Read input";
    name = "control/selector";

/*
    for( const auto& inputObj : props["inputs"].as<JsonArray>() ) {
        addInput(inputObj["name"].as<std::string>());
    }

    for( const auto& inputObj : props["outputs"].as<JsonArray>() ) {
        addOutput(inputObj["name"].as<std::string>());
    }
*/
}

vector<string> Selector::run() {
    vector<string> ret;
    bool update = false;
    /*for (auto& input : inputs) {
      if (input.second) {
        update = true;
        vals[input.first][0] = (input.second);
      }
      input.second = nullptr;
    }*/

    int inpSelect = getValue("SelIn");
    if (inpSelect && update) {
        std::string inpString;
        inpString.push_back(inpSelect + 96);
        value = getInput(inpString);
        setOutput("out", value);
        Serial.print("Select: ");
        Serial.print(inpString.c_str());
    }
    return ret;
}

File: public/nodes/control/selector.js
import Node from "../../node.mjs";
import NodeWork from "../../nodework.mjs";

class Selector extends Node{
    static type = "control/selector";
    static title = "SEL";

    constructor() {
        super();
        this.setProperty("SelIn", "sel", {input: true});
        this.setProperty("a", "a", {input: true});
        this.setProperty("out", "out", {output: true});
    }

    run(update) {
        if (update && isNaN(this.properties.SelIn) == false && this.properties.SelIn > 0) {

            var v = this.properties.SelIn.value;
            this.setOutputData(1, v); // select output feature missing, const 1
            this.properties.SelIn = null
        }
    }

}

NodeWork.registerNodeType(Selector);